base/server/python/pki/server/deployment/scriptlets/infrastructure_layout.py

```diff
@@ -123,7 +124,8 @@ def destroy(self, deployer):
 
         if deployer.mdict['pki_path'] != "/var/lib/pki":
             # remove relocated top-level infrastructure base
-            deployer.directory.delete(deployer.mdict['pki_path'])
+            pki.util.rmtree(deployer.mdict['pki_path'],
+                            deployer.mdict['pki_force_destroy'])
```

Comment:
Does `deployer.mdict['pki_force_destroy']` return a string or boolean? If it returns a string, will string `False` be considered boolean false or true?

Reply:
This is the value from dict: `    'pki_force_destroy': True, `

Reply:
I see. The `pki_force_destroy` and `pki_remove_logs` are actually parsed from command line instead of loaded from pkispawn config. I think we could actually store them directly in the `deployer` object (e.g. `deployer.force` and `deployer.remove_logs`) instead of in `deployer.mdict`, but that's for another time.